main.js

```diff
@@ -0,0 +1,290 @@
+	}
+
+	messageReceivedFromWebSocket(data) {
+		if (this.config.debug_messages) {
+			this.log('debug', `Message received: ${data}`)
+		}
+
+		let msgValue = null
+
+		try {
+			msgValue = JSON.parse(data)
+		} catch (e) {
+			msgValue = data
+		}
+
+		// no sub
+		if (msgValue.msg === 'nosub') {
+			this.log('debug', `Subscription failed: ${msgValue.error.reason}`)
+
+			if (msgValue.error.error === '403') {
+				this.updateStatus(InstanceStatus.BadConfig, `API Key invalid`)
+			}
+		}
+
+		// update collection
+		if (msgValue.msg === 'added') {
+			// process workspaces
+			if (msgValue.collection === 'workspaces') {
+				// add id to fields
+				msgValue.fields.id = msgValue.id
+
+				// add workspace
+				this.cuego.addWorkspace(msgValue.fields)
+
+				if (this.isCueGOInitialized) {
+					this.updateCompanionState()
+				}
+			}
+		}
+
+		// update collection
+		if (msgValue.msg === 'changed') {
+			// process workspaces
+			if (msgValue.collection === 'workspaces') {
+				// add id to fields
+				msgValue.fields.id = msgValue.id
+
+				// add workspace
+				this.cuego.updateWorkspace(msgValue.id, msgValue.fields)
+
+				if (this.isCueGOInitialized) {
+					this.updateCompanionState()
+				}
+			}
+		}
+
+		// update subscription on ready
+		if (msgValue.msg === 'ready') {
+			this.log('debug', `Subscription ready: ${msgValue.subs}`)
+
+			// for each subs update subscription status
+			msgValue.subs.forEach((sub) => {
+				this.cuego.updateSubscription(sub, {
+					ready: true,
+				})
+			})
+
+			this.isCueGOInitialized = true
+		}
+
+		// check if 'api.user.workspaces' is subscribed
+		if (msgValue.msg === 'ready') {
+			if (this.cuego.isInitSubscriptionsComplete()) {
+				this.log('debug', `Lets goooooooo!`)
+
+				this.updateCompanionState()
+			}
+		}
+
+	}
+
+	handlePingPong(data) {
+		let msg = JSON.parse(data)
+
+		if (msg.msg && msg.msg === 'ping') {
+			this.log('debug', `Message sent: {"msg": "pong"}`)
+
+			this.ws.send(JSON.stringify(this.cuego.pong()))
+		}
+	}
+
+	getConfigFields() {
+		return [
+			{
+				type: 'static-text',
+				id: 'info',
+				width: 12,
+				label: 'Information',
+				value: '',
+			},
+			{
+				type: 'textinput',
+				id: 'url',
+				label: 'Target URL',
+				tooltip: 'The URL of the WebSocket server (ws[s]://domain[:port][/path])',
+				width: 12,
+				regex: '/' + this.wsRegex + '/',
+			},
+			{
+				type: 'textinput',
+				id: 'api_key',
+				label: 'API Key',
+				tooltip: 'Connects to the WebSocket server with the given API key',
+				width: 12,
+				regex: '/' + this.apiKeyRegex + '/',
+			},
+			{
+				type: 'checkbox',
+				id: 'reconnect',
+				label: 'Reconnect',
+				tooltip: 'Reconnect on WebSocket error (after 5 secs)',
+				width: 6,
+				default: true,
+			},
+			{
+				type: 'checkbox',
+				id: 'append_new_line',
+				label: 'Append new line',
+				tooltip: 'Append new line (\\r\\n) to cuego',
+				width: 6,
+				default: true,
+			},
```

Comment:
I don't see this being used anywhere (except for the upgrade script). Should it be removed?